lib/logstash/filters/dissect.rb

```diff
@@ -4,43 +4,142 @@
+# `%{?some_field} - named skip field
+
+# Append field notation
+# The value is appended to another value or stored if its the first field seen.
+# The key is prefixed with a `+`.
+# The final value is stored in the Event using the key.
+# The delimiter found before the field or a space is appended before the found value.
+# `%{+some_field}` - an append field
+# `%{+some_field/2}` - and append field with an order modifier.
+# An order modifier, `/number`, allows one to reorder the append sequence.
+# e.g. for a text of `1 2 3 go`, this `%{+a/2} %{+a/1} %{+a/4} %{+a/3}` will build a key/value of `a => 2 1 go 3`
+# Append fields without an order modifier will append in declared order.
+# e.g. for a text of `1 2 3 go`, this `%{a} %{b} %{+a}` will build two key/values of `a => 1 3 go, b => 2`
+
+# Indirect field notation
+# The found value is added to the Event using the found value of another field as the key.
+# The key is prefixed with a `&`.
+# `%{&some_field}` - an indirect field where the key is indirectly sourced from the value of `some_field`.
+# e.g. for a text of `error: some_error, description`, this `error: %{?err}, %{&desc}`will build a key/value of `'an error' => description`
+# Hint: use a Skip field if you do not want the indirection key/value stored.
+# e.g. for a text of `google: 77.98`, this `%{?a}: %{&a}` will build a key/value of `google => 77.98`.
+
+# Note: for append and indirect field the key can refer to a field that already exists in the event before dissection.
+# Note: append and indirect cannot be combined...
+# `%{+&something}` - will add a value to the `&something` key, probably not the intended outcome.
+# `%{&+something}` will add a value to the `+something` key, again unintended.
+
+# Delimiter repetition
+# In the source text if a field has variable width padded with delimiters, the padding will be ignored.
+# e.g. for texts of:
+# ........
+# 00000043 ViewReceiver  I
+# 000000b3 Peer          I
+# ........
+# and a dissection of `%{a} %{b} %{c}`; the padding is ignored.
+#
+# You probably want to put this filter in an if block to ensure that the event
+# contains text with a suitable layout.
+# [source, ruby]
+# filter {
+#   if [type] == "syslog" or "syslog" in [tags] {
+#     dissect {
+#       mapping => {
+#         "message" => "%{timestamp} %{+timestamp} %{+timestamp} %{logsource} %{} %{program}[%{pid}]: %{msg}"
+#       }
+#     }
+#   }
+# }
+
 class LogStash::Filters::Dissect < LogStash::Filters::Base
 
-  # Setting the config_name here is required. This is how you
-  # configure this filter from your Logstash config.
+  config_name "dissect"
+
+  # A hash of dissections of field => value
+  # A later dissection can be done on an earlier one
+  # or they can be independent.
   #
+  # For example
+  # [source, ruby]
   # filter {
-  #   example {
-  #     message => "My message..."
+  #   dissect {
+  #     mapping => {
+  #       "message" => "%{field1} %{field2} %{description}"
```

Comment:
There is no ordering at present, if the value of `event.get("description")` is not a String dissect will do nothing.